autorest/adal/token.go

```diff
@@ -31,8 +32,11 @@ const (
 	// OAuthGrantTypeRefreshToken is the "grant_type" identifier used in refresh token flows
 	OAuthGrantTypeRefreshToken = "refresh_token"
 
-	// managedIdentitySettingsPath is the path to the MSI Extension settings file (to discover the endpoint)
-	managedIdentitySettingsPath = "/var/lib/waagent/ManagedIdentity-Settings"
+	// managedIdentitySettingsLinuxPath is the path to the MSI Extension settings file (to discover the endpoint)
+	managedIdentitySettingsLinuxPath = "/var/lib/waagent/ManagedIdentity-Settings"
```

Comment:
If it were me, and I had two versions of this path to define, I'd employ the same tactics that the standard library took for path separator. There they use OS build tags to get the correct behavior. See here:

https://github.com/golang/go/tree/2d69e9e259ec0f5d5fbeb3498fbd9fed135fe869/src/path/filepath

Reply:
Fixed